prime-router/src/main/kotlin/tokens/AuthenticatedClaims.kt

```diff
@@ -1,135 +1,227 @@
+     * @return true if the request to submit data on behalf of [requiredSender]
+     * is authorized based on the [claims], false otherwise.
      */
-    constructor(
-        _jwtClaims: Map<String, Any>,
-        _organizationNameClaim: String? = null,
-        _isPrimeAdmin: Boolean? = null,
-        _isSenderOrgClaim: Boolean? = null,
-    ) {
-        this.jwtClaims = _jwtClaims
-        this.userName = _jwtClaims[oktaSubjectClaim]?.toString() ?: error("No username in claims")
-        @Suppress("UNCHECKED_CAST")
-        val memberships = jwtClaims[oktaMembershipClaim] as? Collection<String> ?: emptyList()
-        val orgNamePair = extractOrganizationNameFromOktaMembership(memberships)
-        this.isPrimeAdmin = _isPrimeAdmin ?: isPrimeAdmin(memberships)
-        this.organizationNameClaim = _organizationNameClaim ?: orgNamePair?.first // might be null
-        this.isSenderOrgClaim = _isSenderOrgClaim ?: orgNamePair?.second ?: false
+    fun authorizedForSubmission(
+        requiredSender: Sender,
+        request: HttpRequestMessage<String?>
+    ): Boolean {
+        return authorizedForSubmission(requiredSender.organizationName, requiredSender.name, request)
     }
 
     /**
-     * Derive useful info from jwtClaims on [memberships].
-     * @return the *first* well-formed ReportStream organizationName found in [memberships].
-     * (So this won't work if user has many organization claims.)
+     * Determine if these claims authorize access to the /validate and /waters endpoints for
+     * org [requiredOrganization] and optional sender [requiredSender] string.  The [request] is only used for logging.
      *
-     * At the same time, this determines if it's a Sender or Receiver claim.
-     * Returns Pair(organizationName, true) if this organizationName claim is a Sender claim.
-     * Returns Pair(organizationName, false) if this organizationName claim is a Receiver claim.
-     * Returns null if no well-formed organizationName is found in [memberships]
-     * Ignores the PrimeAdmins claim.
+     * @return true if the request to submit data on behalf of the [requiredOrganization] is
+     * authorized based on the [claims], false otherwise.
      */
-    private fun extractOrganizationNameFromOktaMembership(memberships: Collection<String>): Pair<String?, Boolean>? {
-        if (memberships.isEmpty()) return null
-        // examples:   DHSender_ignore, DHPrimeAdmins, DHSender_all-in-one-health-ca, DHaz_phd, DHignore
-        // should return, resp.:  (ignore, true), <nothing>, (all-in-one-health-ca, true, (az_phd, false, (ignore, true)
-        memberships.forEach {
-            if (it == oktaSystemAdminGroup) return@forEach // skip
-            if (it.startsWith(oktaSenderGroupPrefix)) return Pair(it.removePrefix(oktaSenderGroupPrefix), true)
-            if (it.startsWith(oktaGroupPrefix)) return Pair(it.removePrefix(oktaGroupPrefix), false)
+    fun authorizedForSubmission(
```

Comment:
This function "knows" all the scopes that are allowed to do Submission related activities.